cpp/tests/structs/structs_column_tests.cu

```diff
@@ -0,0 +1,441 @@
+
+  vector_of_columns expected_cols_2;
+  expected_cols_2.emplace_back(std::move(expected_bool_col));
+  expected_cols_2.emplace_back(std::move(expected_struct_1));
+  auto expected_struct_2 =
+    cudf::test::structs_column_wrapper(std::move(expected_cols_2), {0, 1, 1, 1, 1, 1}).release();
+
+  cudf::test::expect_columns_equivalent(*expected_struct_2, *struct_2);
+}
+
+TEST_F(StructColumnWrapperTest, StructWithNoMembers)
+{
+  auto struct_col{cudf::test::structs_column_wrapper{}.release()};
+  EXPECT_TRUE(struct_col->num_children() == 0);
+  EXPECT_TRUE(struct_col->null_count() == 0);
+  EXPECT_TRUE(struct_col->size() == 0);
+}
+
+TYPED_TEST(TypedStructColumnWrapperTest, StructsWithMembersWithDifferentRowCounts)
+{
+  auto numeric_col_5 = cudf::test::fixed_width_column_wrapper<TypeParam>{{1, 2, 3, 4, 5}};
+  auto bool_col_4    = cudf::test::fixed_width_column_wrapper<bool>{1, 0, 1, 0};
+
+  EXPECT_THROW(cudf::test::structs_column_wrapper({numeric_col_5, bool_col_4}), cudf::logic_error);
+}
+
+TYPED_TEST(TypedStructColumnWrapperTest, TestListsOfStructs)
+{
+  // Test structs with two members:
+  //  1. Name: String
+  //  2. List: List<TypeParam>
+
+  std::initializer_list<std::string> names = {"Samuel Vimes",
+                                              "Carrot Ironfoundersson",
+                                              "Angua von Uberwald",
+                                              "Cheery Littlebottom",
+                                              "Detritus",
+                                              "Mr Slant"};
+
+  auto num_rows{std::distance(names.begin(), names.end())};
+
+  // `Name` column has all valid values.
+  auto names_col = cudf::test::strings_column_wrapper{names.begin(), names.end()};
+
+  // Numeric column has some nulls.
+  auto ages_col = cudf::test::fixed_width_column_wrapper<TypeParam>{{48, 27, 25, 31, 351, 351},
+                                                                    {1, 1, 1, 1, 1, 0}};
+
+  auto struct_col =
+    cudf::test::structs_column_wrapper({names_col, ages_col}, {1, 1, 1, 0, 0, 1}).release();
+
+  auto expected_unchanged_struct_col = cudf::column(*struct_col);
+
+  auto list_offsets_column =
+    cudf::test::fixed_width_column_wrapper<size_type>{0, 2, 3, 5, 6}.release();
+  auto num_list_rows = list_offsets_column->size() - 1;
+
+  auto list_col = cudf::make_lists_column(num_list_rows,
+                                          std::move(list_offsets_column),
+                                          std::move(struct_col),
+                                          cudf::UNKNOWN_NULL_COUNT,
+                                          {});
+
+  // List of structs was constructed successfully. No exceptions.
+  // Verify that child columns is as it was set.
+
+  cudf::test::expect_columns_equivalent(expected_unchanged_struct_col,
+                                        cudf::lists_column_view(*list_col).child());
+
+#ifndef NDEBUG
```

Comment:
@jrhemstad Is there a cudf standard for debug stuff like this?   This is generally something I like to do but have avoided so far because it doesn't seem to be used in cudf at large.

Reply:
I took my cues from `rolling_test`, for the printing. (Although there, it was under a `#if 0`.)

Reply:
Considering that the cudf debug build doesn't even succeed currently (if you got it to build, I'd love to hear about it), I'm guessing you didn't compile this code?

Reply:
> I'm guessing you didn't compile this code?

Not in debug mode, but I do use the prints. I comment out the `#ifndef NDEBUG`, when I'm debugging. :/

Reply:
I'm happy to remove these prints, if that's preferable.